src/common/framework/query/json_param_value.ts

```diff
@@ -0,0 +1,18 @@
+import { ParamArgument } from '../params_utils.js';
+import { assert } from '../util/util.js';
+
+// JSON can't represent `undefined` and by default stores it as `null`.
+// Instead, store `undefined` as this magic string value in JSON.
+const jsUndefinedMagicValue = '✗undefined';
```

Comment:
I almost made it `𝓊𝓃𝒹𝑒𝒻𝒾𝓃𝑒𝒹` but it was too annoying for encode_selectively 😁

Reply:
heh, nice :) I didn't even know such a thing could be possible